next/blog/pages/blog/index.tsx

```diff
@@ -0,0 +1,145 @@
+                                                : 'border-[#E0E0E0]'
+                                        }`}
+                                        onClick={() => setSelectedCategory(category)}
+                                    >
+                                        {category.toLowerCase()}
+                                    </button>
+                                ))}
+                                <button
+                                    onClick={() => setSelectedCategory('')}
+                                    className={`col-span-2 flex items-center justify-center w-full px-6 py-2.5 rounded-[32px] border transition-colors duration-300 text-xs leading-none tracking-[-0.41px] md:col-span-1 md:col-start-1 md:row-start-1 md:text-lg md:leading-none md:py-3 lg:text-2xl lg:leading-none lg:px-6 lg:py-[14px] ${
+                                        selectedCategory === ''
+                                            ? 'border-appText bg-appText text-white'
+                                            : 'border-[#E0E0E0]'
+                                    }`}
+                                >
+                                    All
+                                </button>
+                            </div>
+
+                        </div>
+            </div>
+                </div>
+                <TopGradient />
+            </div>
+            <div className="container mb-8 md:mb-20 lg:mb-[104px]">
+                {filteredBlogs.length > 0 && selectedCategory ? (
+                    <div className="grid grid-cols-1 gap-6 mb-8 md:grid-cols-2 md:mb-20 lg:grid-cols-3 xl:gap-10 lg:mb-[128px]">
+                        {filteredBlogs.map((blog) => (
+                            <BlogsCard key={blog.slug} blog={blog} />
+                        ))}
+                    </div>
+                ) : (
+                    <div className="leading-none tracking-[-0.41px] text-center mb-8 md:mb-20 lg:text-lg lg:mb-[128px]">
+                        There are no blogs for the applied filter in "{selectedCategory.toLowerCase()}"
+                    </div>
+                )}
+                <div>
+                    <h2 className="leading-none font-medium tracking-[-0.41px] mb-7 md:text-2xl md:leading-none md:mb-10 lg:text-[32px] lg:leading-none lg:mb-12">
+                        All posts
+                    </h2>
+                    <div className="grid grid-cols-1 auto-rows-fr gap-6 p-4 border border-appGray-300 rounded-2xl md:grid-cols-2 lg:grid-cols-3 xl:gap-10 xl:p-8">
+                        {data.blogs.map((blog, index) => (
+                            <BlogsCard key={index} blog={blog} />
+                        ))}
+                    </div>
+                </div>
+            </div>
+        </PageWrapper>
+    );
+}
+
+export const getServerSideProps: GetServerSideProps<APIResponse<BlogsPageData>> = async () => {
```

Comment:
You should use SSG, `getServerSideProps` is used in SSR. For more info. see: https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation